service/src/main/java/bio/terra/workspace/service/resource/referenced/cloud/gcp/ReferencedResourceService.java

```diff
@@ -218,6 +218,14 @@ public ReferencedResource cloneReferencedResource(
             name,
             description);
     // launch the creation flight
-    return createReferenceResource(destinationResource, userRequest);
+    ReferencedResource createdResource = createReferenceResource(destinationResource, userRequest);
+
+    resourceDao.setResourceLineage(
```

Comment:
This should not be done outside of the flight. When you build the reference resource above, that should include the lineage information. Then the lineage can be written or not written as part of the metadata step of the flight. And undone as part of that step.

Reply:
Done